From a1ac3f14117d47f84140dc5b910bdd635ec1baea Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsht@xilinx.com>
Date: Fri, 9 Mar 2018 10:30:47 -0800
Subject: [PATCH] v4l2: Give preference to contiguous format if supported

Currently gstreamer uses single format GST_VIDEO_FORMAT_NV12 for both
NV12 (which uses single contiguous buffer for luma and chroma)
and NV12M (uses two non-contiguous buffers for luma and chroma )
and if device supports both NV12M and NV12 then it gives preference
to NV12M over NV12.

Since the VCU OMX stack currently supports only NV12 so we change
the logic to give preference to NV12 before NV12M whenever GST_VIDEO_FORMAT_NV12
is set.

Upstream Status: Inappropriate [SoC specific]

Signed-off-by: Devarsh Thakkar <devarsht@xilinx.com>
Signed-off-by: Varunkumar Allagadapa <varunkum@xilinx.com>
Signed-off-by: Devarsh Thakkar <devarsht@ti.com>
---
 sys/v4l2/gstv4l2object.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index e5c009b..a403429 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1993,17 +1993,17 @@ gst_v4l2_object_get_caps_info (GstV4l2Object * v4l2object, GstCaps * caps,
   }
 
 
-  /* Prefer the non-contiguous if supported */
-  v4l2object->prefered_non_contiguous = TRUE;
+  /* Prefer the contiguous if supported */
+  v4l2object->prefered_non_contiguous = FALSE;
 
-  if (fourcc_nc)
-    fmt = gst_v4l2_object_get_format_from_fourcc (v4l2object, fourcc_nc);
+  if (fourcc)
+    fmt = gst_v4l2_object_get_format_from_fourcc (v4l2object, fourcc);
   else if (fourcc == 0)
     goto unhandled_format;
 
   if (fmt == NULL) {
-    fmt = gst_v4l2_object_get_format_from_fourcc (v4l2object, fourcc);
-    v4l2object->prefered_non_contiguous = FALSE;
+    fmt = gst_v4l2_object_get_format_from_fourcc (v4l2object, fourcc_nc);
+    v4l2object->prefered_non_contiguous = TRUE;
   }
 
   if (fmt == NULL)
-- 
2.17.1

